lib/sandbox/execute-context.js

```diff
@@ -46,6 +48,9 @@ module.exports = function (scope, code, execution, console, timers, pmapi) {
         // we check if the execution went async by determining the timer queue length at this time
         execution.return.async = (timers.queueLength() > 0);
 
+        // Raise assertion events for the sync tests
+        !_.isEmpty(tests) && legacy.raiseAssertionEvents(execution, pmapi, tests);
```

Comment:
This should be `legacy.finish`(not sure about naming here), that takes care of `raiseAssertionEvents`